src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -54,6 +54,47 @@ public static bool TryGetAnnotationsOfType<T>(this IResource resource, [NotNullW
         }
     }
 
+    /// <summary>
+    /// Attempts to retrieve all annotations of the specified type from the given resource including transitively from parents.
+    /// </summary>
+    /// <typeparam name="T">The type of annotation to retrieve.</typeparam>
+    /// <param name="resource">The resource to retrieve annotations from.</param>
+    /// <param name="result">When this method returns, contains the annotations of the specified type, if found; otherwise, null.</param>
+    /// <returns>true if annotations of the specified type were found; otherwise, false.</returns>
+    public static bool TryGetTransitiveAnnotationsOfType<T>(this IResource resource, [NotNullWhen(true)] out IEnumerable<T>? result) where T : IResourceAnnotation
```

Comment:
nit: This isn't transitive, it's including parents. Transitivity would be the other way.

Reply:
Renamed.